### ISSUE TITLE
Allow Android users to select specific data types "on exit" (parity with Desktop)

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#41155]
### Description

In Brave on Android, when the `Clear data on exit` toggle is "on", the browser will delete all browsing data of all types when the app is fully closed. While this functionality works for many users, it would be useful to have this feature parity what the desktop build does and allow users to select which specific data they want to be cleared when they close the app. 

Additionally, on the implementation, I do notice that when I go to `Settings --> Brave Shields & Privacy --> Clear browsing data...` and tap the check boxes, the boxes that are checked are remembered/persist after leaving that menu. This implies that the app is already keeping track of these specific data types in some way. If we could simply have the browser reference that "list" to see what is or isn't checked when the `Clear data on exit` option is toggled on, this would grant users that additional functionality without needing to implement any changes to the existing UI.

### Steps to reproduce

1. Ensure `Clear data on exit` option is toggled on
2. Close the application
3. Observe that _all_ data is cleared

### Actual result

All data cleared

### Expected result

Ideally only selected data types will be cleared.

### Reproduces how often

Easily reproduced

### Brave version

Any/all

### Device

Any/all

### Channel information

- [X] release (stable)
- [X] beta
- [X] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by Brave-Matt** [comment]
Users still requesting this:
https://github.com/brave/brave-browser/issues/41155
Can pull more requests for this if helpful -- @deeppandya @timchilds any chance we can take another look at this given it's been a good amount of time since the issue was opened initially?

**Comment by ghost** [comment]
It's sad that this receives so less attention despite being requested multiple times across Brave's Discourse Forum, Reddit and other places.

It's also about feature parity across plattforms which I believe to be important for a good user experience, especially for a browser which claims to put the user "first".

**Comment by robbycbennett** [comment]
I agree and it's one of the primary issues that bothers me. However, it's a duplicate of https://github.com/brave/brave-browser/issues/17209

**Comment by chrcoluk** [comment]
Are brave sabotaging themselves on Android or something, this is such an extremely basic feature, it shouldnt even need a request.  Browsers let you customise what is cleared on exit back in the 2000s.

Also delete last browsing data, doesnt remember the previous selection, I am trying to think of why developers would want this behaviour.

**Comment by bsclifton** [comment]
Closing as duplicate of https://github.com/brave/brave-browser/issues/17209
